### ISSUE TITLE
Missile lose lock/retarget needs some love

### ISSUE THREAD

**Issue by royfalk** [vegastrike/Vega-Strike-Engine-Source#328]
Currently, the missile code is in need of refactoring:
- The whole ECM as a random chance needs rework. It should be something more. Lose lock as a factor of range for example.
- Retarget is also in need of attention. We need to specify how and when a missile will lock onto another target.
- Once this is done, open a new issue to actually implement some cool features - lose lock if moving behind another ship. IR to accidentally lock on to another identical ship. Heat seeking to target flares or another ship, etc.
